viewer/vue-client/src/utils/display.js

```diff
@@ -11,25 +11,43 @@ moment.locale('sv');
 export function getDisplayDefinitions(settings) {
   const baseUri = settings.idPath;
   return new Promise((resolve, reject) => {
-    if (settings.mockDisplay === true) {
-      window.lxlInfo('🎭 MOCKING DISPLAY FILE - Using file from local definitions repository');
-      resolve(require('@/../../../../definitions/source/vocab/display.jsonld'));
-    } else {
-      httpUtil.getResourceFromCache(`${baseUri}/vocab/display/data.jsonld`).then((result) => {
-        const clonedResult = cloneDeep(result);
-        each(clonedResult.lensGroups, (lensGroup) => {
-          each(lensGroup.lenses, (lens) => {
-            if (lens.hasOwnProperty('fresnel:extends')) {
-              const [extendLens, extendLevel] = lens['fresnel:extends']['@id'].split('-');
+    const expandInherited = (display) => {
+      const cloned = cloneDeep(display);
+
+      const lensesById = {};
+      each(cloned.lensGroups, (lensGroup) => {
+        each(lensGroup.lenses, (lens) => {
+          if (lens.hasOwnProperty('@id')) {
+            lensesById[lens['@id']] = lens;
+          }
+        });
+      });
+      
+      each(cloned.lensGroups, (lensGroup) => {
+        each(lensGroup.lenses, (lens, key) => {
+          if (lens.hasOwnProperty('fresnel:extends')) {
+            const extendId = lens['fresnel:extends']['@id']; 
+            if (lensesById[extendId]) {
               lens.showProperties.splice(
                 lens.showProperties.indexOf('fresnel:super'),
                 1,
-                ...result.lensGroups[extendLevel].lenses[extendLens].showProperties,
+                lensesById[extendId].showProperties,
               );
+            } else {
+              window.lxlWarning(`👁️ Could not find lens with id '${extendId}' used in 'fresnel:extends' in '${key}'.`);
             }
```

Comment:
This still depends on lenses being in the right order if we have multiple levels of inheritance.
Will rewrite it.